src/config.c

```diff
@@ -235,8 +236,21 @@ config_read(const char *ipath)
    }
   else
    {
-    rha->libexec_path = (char *)malloc(strlen(install_location)+9);
-    if (rha->libexec_path != NULL) sprintf(rha->libexec_path,"%s/libexec",install_location);
+    /* If libexec has a 'blahp' subdirectory, then the blahp files should
+     * be in there.
+     */
+    rha->libexec_path = (char *)malloc(strlen(install_location)+15);
```

Comment:
Really gotta look around a bit to find that the `+15` here refers to strlen("/libexec/blahp"), plus 1 for the NUL byte, which is also big enough for the "/libexec" fallback...  I realize that's kind of a pattern in the blahp code, though i wish we could improve it somehow.